lib/linter.js

```diff
@@ -814,27 +814,35 @@ class Linter extends EventEmitter {
 
             // if this result is from a parseForESLint() method, normalize
             if (parseResult && parseResult.ast) {
-                ast = parseResult.ast;
-            } else {
-                ast = parseResult;
-                parseResult = null;
-            }
 
-            if (ast) {
-                this.sourceCode = new SourceCode(text, ast);
+                // Save all returned values of `parse()` because this source code object can be reused.
+                // This ensures the same result when reused.
```

Comment:
The first argument of `linter#verify` can be a `SourceCode` instance. In that case, parsing is skipped and the `SourceCode` instance is reused.